content/blog/deploying-the-infrastructure-of-oauth-server-for-cms-app/index.md

```diff
@@ -0,0 +1,371 @@
+
+```go
+targetGroupPort := os.Getenv("TARGET_PORT")
+    if targetGroupPort == "" {
+        targetGroupPort = "80"
+    }
+    listenPort := ":" + targetGroupPort
+    fmt.Print("Started running on", listenPort, "\n")
+    // listen on port 80 where we created the target group
+    fmt.Println(http.ListenAndServe(listenPort, nil))
+```
+
+As mentioned previously, `GITHUB_SCOPE` sets the type of access the CMS has for the target repository. It is also an optional stack configuration where the default in [our Pulumi example](https://github.com/pulumi/examples/tree/master/aws-ts-netlify-cms-and-oauth/cms-oauth) is set as `public_repo`.
+
+### Create a Certificate for the OAuth Server
+
+The process is similar to how we created a certificate for CMS in our [previous post]({{< relref "/blog/deploying-netlify-cms-on-aws" >}}).
+
+We can extract what we need just for creating the certificate and delete irrelevant parts including uploading content to the S3 bucket and configure CloudFront. Then the rest only involves creating an east region provider, a certificate, and certificate validation.
+
+When we create the Alias Record, we configure it with application load balancer settings.
+
+```typescript
+// Creates a new Route53 DNS record pointing the domain to the CloudFront distribution.
+function createAliasRecord(
+    targetDomain: string, lb: awsx.elasticloadbalancingv2.ApplicationLoadBalancer): aws.route53.Record {
+    const domainParts = getDomainAndSubdomain(targetDomain);
+    const hostedZoneId = aws.route53.getZone({ name: domainParts.parentDomain }, { async: true }).then(zone => zone.zoneId);
+    return new aws.route53.Record(
+        targetDomain,
+        {
+            name: domainParts.subdomain,
+            zoneId: hostedZoneId,
+            type: "A",
+            aliases: [
+                {
+                    name: lb.loadBalancer.dnsName,
+                    zoneId: lb.loadBalancer.zoneId,
+                    evaluateTargetHealth: true,
+                },
+            ],
+        });
+}
+// Create the aliasRecord with targetdomain and application load balancer
+const aRecord = createAliasRecord(cmsStackConfig.targetDomain, alb);
+```
+
+We run `pulumi up` to deploy.
+
+### Github Workflow (Optional)
+
+We can automate our build with the Github workflow in the folder `cms-oauth/.github/workflows/build-and-deploy.yml`. The workflow uses the repository secret we set in Github.
+
+![Github Secret](./github_secret.jpg)
+
+## Last Step
+
+Finally, we need to update the CMS configuration file.
+
+To combine our deployment of CMS and CMS OAuth Server, we specify the site_domain and URL in `./public/config.yml` of the CMS folder.
+
+```yaml
+backend:
+  name: github
+  # Replace this with the Github repo you want to make change
+  repo: github-username/target-github-repo
+  # This site_domain and base_url are sudo domains
+  # Replace site_domain with targetDomain pulumi stack configuration inside cms/infrastructure folder
```

Comment:
I don't understand this comment (and the one on the next line too). Could you please take a look at this and perhaps rephrase it?